airflow/contrib/operators/gcp_cloud_build_operator.py

```diff
@@ -0,0 +1,177 @@
+
+        if not isinstance(source, six.string_types):
+            return
+
+        self.body["source"]["storageSource"] = self._convert_storage_url_to_dict(source)
+
+    def process_body(self):
+        self._verify_source()
+        self._reformat_source()
+        return self.body
+
+    @staticmethod
+    def _convert_repo_url_to_dict(source):
+        """
+        Convert url to repository in Google Cloud Source to a format supported by the API
+
+        Example valid input:
+
+        .. code-block:: none
+            https://source.developers.google.com/p/airflow-project/r/airflow-repo#branch-name
+
+        """
+        url_parts = urlparse(source)
+
+        match = REGEX_REPO_PATH.search(url_parts.path)
+
+        if url_parts.scheme != "https" or url_parts.hostname != "source.developers.google.com" or not match:
+            raise AirflowException(
+                "Invalid URL. You must pass the URL in the format: "
+                "https://source.developers.google.com/p/airflow-project/r/airflow-repo#branch-name"
+            )
+
+        project_id = unquote(match.group("project_id"))
+        repo_name = unquote(match.group("repo_name"))
+
+        source_dict = {"projectId": project_id, "repoName": repo_name, "branchName": "master"}
+
+        if url_parts.fragment:
+            source_dict["branchName"] = url_parts.fragment
+
+        return source_dict
+
+    @staticmethod
+    def _convert_storage_url_to_dict(storage_url):
+        """
+        Convert url to object in Google Cloud Storage to a format supported by the API
+
+        Example valid input:
+
+        .. code-block:: none
+            gs://bucket-name/object-name.tar.gz
+
+        """
+        url_parts = urlparse(storage_url)
+
+        if url_parts.scheme != "gs" or not url_parts.hostname or not url_parts.path or url_parts.path == "/":
+            raise AirflowException(
+                "Invalid URL. You must pass the URL in the format: "
+                "gs://bucket-name/object-name.tar.gz#24565443"
+            )
+
+        source_dict = {"bucket": url_parts.hostname, "object": url_parts.path[1:]}
+
+        if url_parts.fragment:
+            source_dict["generation"] = url_parts.fragment
+
+        return source_dict
+
+
+class CloudBuildCreateBuildOperator(BaseOperator):
+    """
+    Starts a build with the specified configuration.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:CloudBuildCreateBuildOperator`
+
+    :param body: The request body.
+        See: https://cloud.google.com/cloud-build/docs/api/reference/rest/Shared.Types/Build
+    :type body: dict
+    :param gcp_conn_id: The connection ID to use to connect to Google Cloud Platform.
+    :type gcp_conn_id: str
+    :param api_version: API version used (for example v1 or v1beta1).
+    :type api_version: str
+    """
+
+    # [START gcf_function_delete_template_fields]
```

Comment:
I think this flag's name is a copy-paste leftower

Reply:
I updated a PR. Currently, these markers are not needed in the code.